pkg/inject/sidecar_builder.go

```diff
@@ -190,6 +191,14 @@ func buildEnvoySidecar(vars EnvoyTemplateVariables, env map[string]string) (core
 		},
 	}
 
+	if vars.WaitUntilProxyStarts {
+		envoy.Lifecycle.PostStart = &corev1.Handler{
+			Exec: &corev1.ExecAction{Command: []string{
+				"sh", "-c", fmt.Sprintf("until curl -s http://localhost:%d/server_info | grep state | grep -q LIVE; do sleep 1; done", vars.AdminAccessPort),
```

Comment:
Is there no simple CLI command in [agent](https://aws.amazon.com/about-aws/whats-new/2022/02/aws-app-mesh-agent-envoy/) to do this? We should remove curl from the image and then this will break. @abaptiste can you guide us here.

Reply:
curl is something we already use in our readiness probe https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/pkg/inject/sidecar_builder.go#L224, however, i agree with your point, because im actually running into an issue where sleep is not found in the envoy container (maybe because of path issues, something im still investigating). if there are alternative executables or agent commands you referenced that are better, pls let me know. is there more information on agent? I followed that link but it doesnt say much

Reply:
To add to @kiranmeduri 's point, the concern is the vendor upstream of us (agent) controls the container image we are leveraging, so they could break us unintentionally or without us knowing. Even if the agent team vends an _isReady_  (example) script for us to invoke (which in turn calls curl), at least there is a formal contract between us that they cannot break. Or we need to be able to enforce the requirement that _curl_ be present in the image somehow. I don't know if there is a way to do the latter.  _sleep_ is a related but different issue IMHO. I guess they could ship with no shell at all, but I think sleep is _ususally_ available in busybox or even the most chroot'd jail ?

Reply:
I think the curl concern is reasonable but given that dependency already exists in the controller It seems to me it shouldn't hold up this change. We should maybe spin up a separate ticket with the agent folks to get these scripts encapsulated as @ar4096 suggests.

Reply:
Sorry I'm late to the PR.  The `curl` binary in the image is our replacement of the standard tool.  It's a static golang binary that replaces the traditional binary to eliminate quite a few system (34) libraries.

Secondly, the binaries present in the image do not include sleep or timeout.

I would recommend leveraging the agent itself to perform the check since it's responsible for starting Envoy and already does health checks of the proxy.  You could build a equivalent behavior of `until curl -s ...` into the supervisor  binary itself and define some IPC between the process running Envoy and the command line invocation to get the proxy's state.

Reply:
hi @abaptiste, thx for the comment. what is the supervisor binary here you are referring to? also i looked at the agent codebase and im a bit fuzzy on what you are recommending in regards to the command line invocation. do you mind elaborating where that would live? also if we do go down this path, as @kiranmeduri mentioned here in the quip design comment https://quip-amazon.com/5JRzAMebdaEO#IOK9DA9Cpzk, this would couple the controller release with the envoy image release. what is the release cadence of the agent binary? how much time should we estimate to release the coupled agent binary changes?

Reply:
I was able to catch up with @abaptiste about his recommendation. the next step is to discuss with @abaptiste and @ar4096 about the potential solutions and come to a consensus. i will follow-up in this thread with the decision we end up with.